wgpu-hal/src/dx12/device.rs

```diff
@@ -355,84 +369,64 @@ impl crate::Device<super::Api> for super::Device {
             Flags: conv::map_buffer_usage_to_resource_flags(desc.usage),
         };
 
-        let is_cpu_read = desc.usage.contains(crate::BufferUses::MAP_READ);
-        let is_cpu_write = desc.usage.contains(crate::BufferUses::MAP_WRITE);
-
-        let heap_properties = d3d12::D3D12_HEAP_PROPERTIES {
-            Type: d3d12::D3D12_HEAP_TYPE_CUSTOM,
-            CPUPageProperty: if is_cpu_read {
-                d3d12::D3D12_CPU_PAGE_PROPERTY_WRITE_BACK
-            } else if is_cpu_write {
-                d3d12::D3D12_CPU_PAGE_PROPERTY_WRITE_COMBINE
-            } else {
-                d3d12::D3D12_CPU_PAGE_PROPERTY_NOT_AVAILABLE
-            },
-            MemoryPoolPreference: match self.private_caps.memory_architecture {
-                super::MemoryArchitecture::NonUnified if !is_cpu_read && !is_cpu_write => {
-                    d3d12::D3D12_MEMORY_POOL_L1
-                }
-                _ => d3d12::D3D12_MEMORY_POOL_L0,
-            },
-            CreationNodeMask: 0,
-            VisibleNodeMask: 0,
-        };
-
-        let hr = unsafe {
-            self.raw.CreateCommittedResource(
-                &heap_properties,
-                if self.private_caps.heap_create_not_zeroed {
-                    D3D12_HEAP_FLAG_CREATE_NOT_ZEROED
-                } else {
-                    d3d12::D3D12_HEAP_FLAG_NONE
-                },
-                &raw_desc,
-                d3d12::D3D12_RESOURCE_STATE_COMMON,
-                ptr::null(),
-                &d3d12::ID3D12Resource::uuidof(),
-                resource.mut_void(),
-            )
-        };
+        let (hr, allocation) = my_buffer_hr(self, desc, raw_desc, &mut resource)?;
```

Comment:
hr?

Reply:
HRESULT, I just copied what the name was before. Looks like I forgot to rename the buffer and texture resource creation functions.